sw/device/tests/entropy_src_bypass_mode_health_test.c

```diff
@@ -0,0 +1,487 @@
+  TRY(dif_entropy_src_configure(&entropy_src, config, kDifToggleEnabled));
+
+  return OK_STATUS();
+}
+
+status_t configure_stringent_health_tests(void) {
+  LOG_INFO("Configuring stringent health test thresholds...");
+
+  // Ensure the entropy source is not locked
+  bool is_locked;
+  TRY(dif_entropy_src_is_locked(&entropy_src, &is_locked));
+  TRY_CHECK(!is_locked,
+            "Entropy source is locked. Cannot configure ENTROPY_SRC");
+
+  // Configure Repetition Count Test with stringent threshold
+  dif_entropy_src_health_test_config_t repcnt_test_config = {
+      .test_type = kDifEntropySrcTestRepetitionCount,
+      .high_threshold = 1,
+      .low_threshold = 0,
+  };
+  TRY(dif_entropy_src_health_test_configure(&entropy_src, repcnt_test_config));
+
+  // Configure Adaptive Proportion Test with stringent threshold
+  dif_entropy_src_health_test_config_t adaptp_test_config = {
+      .test_type = kDifEntropySrcTestAdaptiveProportion,
+      .high_threshold = 1,
+      .low_threshold = 0,
+  };
+  TRY(dif_entropy_src_health_test_configure(&entropy_src, adaptp_test_config));
+
+  // Configure Bucket Test with stringent threshold
+  dif_entropy_src_health_test_config_t bucket_test_config = {
+      .test_type = kDifEntropySrcTestBucket,
+      .high_threshold = 1,
+      .low_threshold = 0,
+  };
+  TRY(dif_entropy_src_health_test_configure(&entropy_src, bucket_test_config));
+
+  // Configure Markov Test with stringent threshold
+  dif_entropy_src_health_test_config_t markov_test_config = {
+      .test_type = kDifEntropySrcTestMarkov,
+      .high_threshold = 1,
+      .low_threshold = 0,
+  };
+  TRY(dif_entropy_src_health_test_configure(&entropy_src, markov_test_config));
+
+  return OK_STATUS();
+}
+
+status_t verify_recoverable_alert(void) {
+  LOG_INFO("Verifying recoverable alerts...");
+
+  // Retrieve the recoverable alerts
+  uint32_t alerts;
+  TRY(dif_entropy_src_get_recoverable_alerts(&entropy_src, &alerts));
+
+  if (alerts != 0) {
+    LOG_INFO("Recoverable alert detected. Alerts: 0x%x", alerts);
+
+    // Define alert bit masks based on documentation
+    const uint32_t kEsMainSmAlertBit = 1 << 12;
+    const uint32_t kEsThreshCfgAlertBit = 1 << 14;
+
+    // Debug check if ES_THRESH_CFG_ALERT is set
+    if (alerts & kEsThreshCfgAlertBit) {
+      LOG_ERROR("ES_THRESH_CFG_ALERT is asserted");
+      return INTERNAL();
+    } else {
+      LOG_INFO("ES_THRESH_CFG_ALERT is not asserted and expected");
+    }
+
+    // Check if ES_MAIN_SM_ALERT is set
+    if (alerts & kEsMainSmAlertBit) {
+      LOG_INFO("ES_MAIN_SM_ALERT is asserted as expected");
+    } else {
+      LOG_ERROR("ES_MAIN_SM_ALERT is not asserted when expected");
+      return INTERNAL();
+    }
+
+    // Clear the recoverable alerts
+    TRY(dif_entropy_src_clear_recoverable_alerts(&entropy_src, alerts));
+
+    // Verify that the alerts have been cleared
+    TRY(dif_entropy_src_get_recoverable_alerts(&entropy_src, &alerts));
+
+    if (alerts == 0) {
```

Comment:
You also could replace this check with `TRY_CHECK()`.

Reply:
Noted!